jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/ProxyProvider.java

```diff
@@ -18,33 +18,64 @@
 
 import com.google.common.annotations.VisibleForTesting;
 import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.settings.Proxy;
+import org.apache.maven.settings.Settings;
+import org.apache.maven.settings.building.SettingsProblem;
+import org.apache.maven.settings.crypto.DefaultSettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecrypter;
+import org.apache.maven.settings.crypto.SettingsDecryptionRequest;
+import org.apache.maven.settings.crypto.SettingsDecryptionResult;
 
 /** Propagates proxy configuration from Maven settings to system properties. */
 class ProxyProvider {
 
   private static final ImmutableList<String> PROXY_PROPERTIES =
       ImmutableList.of("proxyHost", "proxyPort", "proxyUser", "proxyPassword");
 
+  private static final ImmutableList<String> PROXY_PROTOCOLS = ImmutableList.of("https", "http");
   /**
    * Initializes proxy settings based on Maven settings.
    *
    * @param settings Maven settings
    */
-  static void init(DecryptedMavenSettings settings) {
-    configureProxy(settings, "https");
-    configureProxy(settings, "http");
-  }
+  static void populateSystemProxyProperties(Settings settings, SettingsDecrypter decrypter)
+      throws MojoExecutionException {
+    List<Proxy> proxies =
+        PROXY_PROTOCOLS
```

Comment:
How about having this as a local var instead of having a field? It gives me a false impression that I can readily add other protocols, but it really must be either `http` or `https` to form `http.proxyHost` or `https.proxyHost`.